pkg/templates/danglingservicemonitor/template.go

```diff
@@ -0,0 +1,79 @@
+package danglingservicemonitor
+
+import (
+	"fmt"
+
+	k8sMonitoring "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/danglingservicemonitor/internal/params"
+	v1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Dangling Service Monitor",
+		Key:         "dangling-servicemonitor",
+		Description: "Flag service monitors which do not match any service",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.ServiceMonitor},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				serviceMonitor, ok := object.K8sObject.(*k8sMonitoring.ServiceMonitor)
+				if !ok {
+					return nil
+				}
+				nsSelector := serviceMonitor.Spec.NamespaceSelector
+				labelSelectors := serviceMonitor.Spec.Selector.MatchLabels
+				if len(labelSelectors) == 0 && len(nsSelector.MatchNames) == 0 && !nsSelector.Any {
+					return []diagnostic.Diagnostic{{
+						Message: "service monitor has no selector specified",
+					}}
+				}
+				labelSelector, err := metaV1.LabelSelectorAsSelector(&metaV1.LabelSelector{MatchLabels: serviceMonitor.Spec.Selector.MatchLabels})
+				if err != nil {
+					return []diagnostic.Diagnostic{{
+						Message: fmt.Sprintf("service monitor has invalid label selector: %v", err),
+					}}
+				}
+				for _, obj := range lintCtx.Objects() {
+					services, found := obj.K8sObject.(*v1.Service)
+					if !found {
+						continue
+					}
+					if checkNamespaceSelector(nsSelector, services) && !p.DisableNamespaceSelectorCheck {
+						return nil
+					}
+					if labelSelector.Matches(labels.Set(services.Labels)) && len(labelSelectors) != 0 && !p.DisableLabelSelectorCheck {
+						// Found!
+						return nil
+					}
+
+				}
+				return []diagnostic.Diagnostic{{Message: fmt.Sprintf("no services found matching Service Monitor's labelSelector (%s) or namespace selector (%s)", labelSelector, nsSelector.MatchNames)}}
+			}, nil
+		}),
+	})
+}
+
+func checkNamespaceSelector(namespaceSelector k8sMonitoring.NamespaceSelector, service *v1.Service) bool {
+	if namespaceSelector.Any {
+		return true
+	} else if len(namespaceSelector.MatchNames) != 0 {
+		for _, ns := range namespaceSelector.MatchNames {
+			if ns == service.Namespace {
+				return true
+			}
+		}
+	}
+	return false
```

Comment:
`else if len(namespaceSelector.MatchNames) != 0` is redundant 
```suggestion
	if namespaceSelector.Any {
		return true
	}
	for _, ns := range namespaceSelector.MatchNames {
		if ns == service.Namespace {
			return true
		}
	}
	return false
```